### PR TITLE
fix(webpack-dev-server): Pass in jsconfig to get webpack config for Next.js >= 13.2.1

### LINKED ISSUE #25951: Component testing broken with NextJS 13.2.1
### Current behavior

Component tests run fine on NextJS v13.1.1, but fail to compile on v13.2.1

The error received is unable to find module, where the module resolution relies on the `tsconfig` basePath setting in `/cypress/tsconfig.json`

Component test cypress config settings:

```
component: {
    devServer: {
      framework: 'next',
      bundler: 'webpack',
    },
}
```

### Desired behavior

Component tests should compile and run using the `tsconfig.json` settings.

Possibly related NextJS change/update: https://github.com/vercel/next.js/pull/45455

### Test code to reproduce

Run any component test that uses a custom cypress `tsconfig` with NextJS v13.2.1

### Cypress Version

latest

### Node version

v16.3

### Operating System

MacOSX And Github Actions Linux

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by astone123**
- Closes #25951

### Additional details

With the release of Next.js 13.2.1, we need to pass the JS/TS config into `getBaseWebpackConfig`, otherwise the config won't be respected when running the dev server. This PR adds some logic to get the user's JS/TS configuration and pass it to `getBaseWebpackConfig`.

### Steps to test

1. Create a Next.js project (v13.2.1) that uses the `src/` dir (it's in the generator options)
2. Configure Cypress for CT
3. Have a tsconfig.json file in the project root with these options
```js
{
  "compilerOptions": {
    // ...
    "paths": {
      "*": ["src/*"]
    },
  }
  "baseUrl": '.'
}
```
4. In `cypress/support/component.ts`, import a stylesheet like this
```js
import 'styles/globals.css'
```
5. Run a component test and verify that Webpack is able to resolve this import

Also test on < v13.2.1 and make sure that everything still works. 

### PR Tasks

- [ ] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by astone123** [comment]
Thoughts about testing this with a system test? I looked into it for a while but was having some problems and started questioning whether or not it would be worth the effort. I was able to test it pretty well manually using some example projects.

File: npm/webpack-dev-server/src/helpers/nextHandler.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ function getNextJsPackages (devServerConfig: WebpackDevServerConfig) {
     const getNextJsBaseWebpackConfigPath = require.resolve('next/dist/build/webpack-config', resolvePaths)
 
     packages.getNextJsBaseWebpackConfig = require(getNextJsBaseWebpackConfigPath).default
+    packages.nextLoadProjectInfo = require(getNextJsBaseWebpackConfigPath).loadProjectInfo
```
Comment (ZachJW34) [inline npm/webpack-dev-server/src/helpers/nextHandler.ts]:
```suggestion
    packages.getNextJsBaseWebpackConfig = require(getNextJsBaseWebpackConfigPath).default
    packages.nextLoadProjectInfo = require(getNextJsBaseWebpackConfigPath).loadProjectInfo
    
    const { default: getNextJsBaseWebpackConfig, loadProjectInfo } = require('getNextJsBaseWebpackConfigPath')
```

2 requires for the price of one

File: npm/webpack-dev-server/src/helpers/nextHandler.ts [hunk 2]
```diff
@@ -131,13 +132,49 @@ function getNextJsPackages (devServerConfig: WebpackDevServerConfig) {
...
-  const { loadConfig, getNextJsBaseWebpackConfig } = getNextJsPackages(devServerConfig)
+  const { loadConfig, getNextJsBaseWebpackConfig, nextLoadProjectInfo } = getNextJsPackages(devServerConfig)
 
   const nextConfig = await loadConfig('development', devServerConfig.cypressConfig.projectRoot)
   const runWebpackSpan = getRunWebpackSpan(devServerConfig)
   const reactVersion = getReactVersion(devServerConfig.cypressConfig.projectRoot)
+  // If nextLoadProjectInfo is not a function, then the Next.js version is < 13.2.1
+  const projectInfo = typeof nextLoadProjectInfo === 'function' ?
+    await nextLoadProjectInfo({ dir: devServerConfig.cypressConfig.projectRoot, config: nextConfig, dev: true }) :
+    undefined
```
Comment (ZachJW34) [inline npm/webpack-dev-server/src/helpers/nextHandler.ts]:
```suggestion
const projectInfo = await nextLoadProjectInfo?.({ dir: devServerConfig.cypressConfig.projectRoot, config: nextConfig, dev: true })
```

Could one line it with optional access

**Review by ZachJW34 — COMMENTED** [review]
I think a system-test covering this would be beneficial

**Review by ZachJW34 — APPROVED** [review]

**Comment by marktnoonan** [comment]
I'm following the directions to test and trying a couple of things, but I consistently get 

```
 Cannot find module '@/styles/globals.css'
```

I wonder if I am missing some little detail.

File: npm/webpack-dev-server/src/helpers/nextHandler.ts [hunk 3]
```diff
@@ -131,13 +134,46 @@ function getNextJsPackages (devServerConfig: WebpackDevServerConfig) {
...
+  ]
  */
 async function loadWebpackConfig (devServerConfig: WebpackDevServerConfig): Promise<Configuration> {
-  const { loadConfig, getNextJsBaseWebpackConfig } = getNextJsPackages(devServerConfig)
+  const { loadConfig, getNextJsBaseWebpackConfig, nextLoadProjectInfo } = getNextJsPackages(devServerConfig)
 
   const nextConfig = await loadConfig('development', devServerConfig.cypressConfig.projectRoot)
   const runWebpackSpan = getRunWebpackSpan(devServerConfig)
   const reactVersion = getReactVersion(devServerConfig.cypressConfig.projectRoot)
+  const projectInfo = await nextLoadProjectInfo?.({ dir: devServerConfig.cypressConfig.projectRoot, config: nextConfig, dev: true })
```
Comment (marktnoonan) [inline npm/webpack-dev-server/src/helpers/nextHandler.ts]: Let's add a comment explaining the `dev: true`?

**Review by marktnoonan — APPROVED** [review]
Confirmed this works as expected for me. +1 to a system test

File: npm/webpack-dev-server/src/helpers/nextHandler.ts (hunk 3 above)
Comment (astone123) [reply npm/webpack-dev-server/src/helpers/nextHandler.ts]: I updated this so that we use `loadJsConfig` directly which doesn't need the `dev` param

**Review by jordanpowell88 — APPROVED** [review]

**Pull request merged**